src/main/java/com/company/mvcfreemarker/controller/UserController.java

```diff
@@ -0,0 +1,23 @@
+package com.company.mvcfreemarker.controller;
+
+import com.company.mvcfreemarker.service.UserService;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+@Controller
+@RequestMapping("/portal/users")
+public class UserController {
+    private UserService userService;
```

Comment:
I don't think we need a lot of separate service classes in the example. The main purpose of the example is to show the MVC part, so let's focus on it. Let's create a method in the controller, say, `getAllUserDtoList()`. Inside the method invoke the DataManager and convert the result list into the list of DTOs. I think this will be enough

Reply:
Ok. I just want to do according to best practices. For this case: dependency inversion, single responsibility and etc.